File: tutorials/lang-tut/lua/calc.lua

iter = 20000

fib_cache = {}

function fib(n)
    if n == 0 or n == 1 then
        return 1
    end
    if not fib_cache[n-1] then
        fib_cache[n-1] = fib(n-1)
    end
    if not fib_cache[n-2] then
        fib_cache[n-2] = fib(n-2)
    end

    return fib_cache[n-1] + fib_cache[n-2]
end


function calculate(iter)
    local res = 0
    for i=1,iter do
        local v = fib(i)
        for j=1,iter do
            res = res + (v % fib(j))
        end
    end
    return res
end

print(fib(100))

res = calculate(iter)
print(res)

File: tutorials/lang-tut/lua/calc.c
#include <stdio.h>

int fib(int n)
{
    if(n == 1 || n == 0)
        return 1;
    return fib(n-1) + fib(n-2);
}

int calculate(int n)
{
    int j = 0;
    for(int i=0;i<n;i++) {
        j += fib(25) * i;
    }

    return j;
}

int main(int argc, char** argv)
{
    int iter = 600000;
    int res = calculate(iter);
    printf("%d\n", res);
    return 0;
}


